test/specs/visualforceLsp.e2e.ts

```diff
@@ -0,0 +1,136 @@
+      1
+    );
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(
+      workbench,
+      'Running SFDX: Push Source to Default Org and Override Conflicts',
+      fiveMinutes
+    );
+    const successPushNotificationWasFound = await utilities.notificationIsPresent(
+      workbench,
+      'SFDX: Push Source to Default Org and Override Conflicts successfully ran'
+    );
+    expect(successPushNotificationWasFound).toBe(true);
+  });
+
+  step('Verify Extension is Running', async () => {
+    utilities.log(
+      `${testSetup.testSuiteSuffixName} - Verify Extension is Running`
+    );
+    // Using the Command palette, run Developer: Show Running Extensions
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'Developer: Show Running Extensions',
+      1
+    );
+
+    // Close panel so the visualforce extension can be seen
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'View: Close Panel',
+      1
+    );
+
+    // Verify Visualforce extension is present and running
+    const extensionNameDivs = await $$('div.name');
+    let extensionWasFound = false;
+    for (const extensionNameDiv of extensionNameDivs) {
+      const text = await extensionNameDiv.getText();
+      if (text.includes('salesforce.salesforcedx-vscode-visualforce')) {
+        extensionWasFound = true;
+      }
+    }
+    expect(extensionWasFound).toBe(true);
+  });
+
+  step('Go to Definition', async () => {
+    utilities.log(`${testSetup.testSuiteSuffixName} - Go to Definition`);
+    // Get open text editor
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+    const textEditor = (await editorView.openEditor(
+      'FooPage.page'
+    )) as TextEditor;
+    await textEditor.moveCursor(1, 25);
+
+    // Go to definition through F12
+    await browser.keys(['F12']);
+    await utilities.pause(1);
+
+    // TODO: go to definition is actually not working
+
+    // // Verify 'Go to definition' took us to the definition file
+    // const activeTab = await editorView.getActiveTab();
+    // const title = await activeTab?.getTitle();
+    // expect(title).toBe('MyController.cls');
+  });
+
+  step('Autocompletion', async () => {
+    utilities.log(`${testSetup.testSuiteSuffixName} - Autocompletion`);
+    // Get open text editor
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+    const textEditor = (await editorView.openEditor(
+      'FooPage.page'
+    )) as TextEditor;
+    await textEditor.typeTextAt(3, 1, '\t<apex:pageM');
+    await utilities.pause(1);
+
+    // Verify autocompletion options are present
+    const autocompletionOptions = await $$(
+      'textarea.inputarea.monaco-mouse-cursor-text'
+    );
+    expect(await autocompletionOptions[0].getAttribute('aria-haspopup')).toBe(
```

Comment:
I'm not a fan of this style, since it makes debugging more difficult.  I suggest splitting it up and changing to:
```
    const ariaHasPopupAttribute = await autocompletionOptions[0]
      .getAttribute('aria-haspopup');
    expect(ariaHasPopupAttribute).toBe('true');

    const ariaAutocompleteAttribute = await autocompletionOptions[0]
      .getAttribute('aria-autocomplete');
    expect(ariaAutocompleteAttribute).toBe('list');
```